internal/services/recoveryservices/backup_policy_vm_resource.go

```diff
@@ -517,6 +538,24 @@ func expandBackupProtectionPolicyVMRetentionWeeklyFormat(block map[string]interf
 	return &weekly
 }
 
+func flattenBackupProtectionPolicyVMResourceGroup(rpDetail *backup.InstantRPAdditionalDetails) []interface{} {
+	if rpDetail == nil || rpDetail.AzureBackupRGNamePrefix == nil {
+		return nil
+	}
```

Comment:
Do we need to nil check here since we're already nil checking the input on https://github.com/hashicorp/terraform-provider-azurerm/blob/2186bdcdbdbc3ff18fd75698628dc9bcc11d7c3c/internal/services/recoveryservices/backup_policy_vm_resource.go#L275